File: javascript/fibonacci/fibonacci.js

function fibonacci(n) {
    if (n == 0 || n == 1) {
        return n;
    } else {
        return fibonacci(n-1) + fibonacci(n-2);
    }
}

function fibonacciLine(n) {
    var list = new Array;
    for (var x = 0; x < n; x++) {
        list.push(fibonacci(x));
    }
    return list;
}

print(fibonacciLine(10));

File: ruby/fibonacci/fibonacci.rb

def fibonacci(n)
    if n == 0 or n == 1
        return n
    else
        return fibonacci(n-1) + fibonacci(n-2)
    end
end

def fibonacciLine(n)
    list = Array.new
    for i in (0..n)
        list << fibonacci(i)
    end
    return list
end

puts fibonacciLine(10)
